### PR TITLE
Add support for The Graph staking in Arbitrum One network

### PR DESCRIPTION
**Pull request by moxw**
Related to #7358 

sqldiff of `data/global.db`
```sql
INSERT INTO assets(rowid,identifier,name,type) VALUES(6852,'eip155:42161/erc20:0x9623063377AD1B27544C965cCd7342f7EA7e88C7','Graph Token','C');
INSERT INTO common_asset_details(rowid,identifier,symbol,coingecko,cryptocompare,forked,started,swapped_for) VALUES(6852,'eip155:42161/erc20:0x9623063377AD1B27544C965cCd7342f7EA7e88C7','GRT','the-graph',NULL,NULL,1669815527,NULL);
INSERT INTO evm_tokens(rowid,identifier,token_kind,chain,address,decimals,protocol) VALUES(5920,'eip155:42161/erc20:0x9623063377AD1B27544C965cCd7342f7EA7e88C7','A',42161,'0x9623063377AD1B27544C965cCd7342f7EA7e88C7',18,NULL);
INSERT INTO multiasset_mappings(rowid,collection_id,asset) VALUES(678,45,'eip155:42161/erc20:0x9623063377AD1B27544C965cCd7342f7EA7e88C7');
```

### CODE REVIEW

File: rotkehlchen/chain/ethereum/modules/thegraph/constants.py [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
 from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.evm.types import string_to_evm_address
 
 CPT_THEGRAPH: Final = 'thegraph'
-
 THEGRAPH_CPT_DETAILS = CounterpartyDetails(
     identifier=CPT_THEGRAPH,
     label='The Graph',
     image='thegraph.svg',
 )
```
Comment (yabirgb) [inline rotkehlchen/chain/ethereum/modules/thegraph/constants.py:10]: I didn't notice before but the counterpart is duplicated. You can remove it from here and use the constatn in evm

Comment (yabirgb) [reply rotkehlchen/chain/ethereum/modules/thegraph/constants.py:10]: same for the `CPT_THEGRAPH` string

File: rotkehlchen/chain/evm/decoding/thegraph/decoder.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+# example delegate() call: https://etherscan.io/tx/0x6ed3377db652151fb8e4794dd994a921a2d029ad317bd3f2a2916af239490fec
+TOPIC_STAKE_DELEGATED: Final = b'\xcd\x03f\xdc\xe5$}\x87O\xfc`\xa7b\xaaz\xbb\xb8,\x16\x95\xbb\xb1q`\x9c\x1b\x88a\xe2y\xebs'  # noqa: E501
+# example undelegate() call: https://etherscan.io/tx/0x5ca5244868d9c0d8c30a1cad0feaf137bd28acd9c3f669a09a3a199fd75ad25a
+# example undelegate() call: https://arbiscan.io/tx/0xaa1d02b966d38fb7ade7768eae3dafc4bd4948090eadbb72f51fe6bab5f099cc
+TOPIC_STAKE_DELEGATED_LOCKED: Final = b'\x040\x18?\x84\xd9\xc4P#\x86\xd4\x99\xda\x80eC\xde\xe1\xd9\xde\x83\xc0\x8b\x01\xe3\x9am!\x16\xc4;%'  # noqa: E501
```
Comment (yabirgb) [inline rotkehlchen/chain/evm/decoding/thegraph/decoder.py:36]: is there something special about the arb example here? or did you frogot to remove it?

File: rotkehlchen/tests/unit/decoders/test_thegraph.py [hunk 3]
```diff
@@ -163,14 +170,189 @@ def test_thegraph_delegated_withdrawn(database, ethereum_inquirer):
...
+            timestamp=timestamp,
+            location=Location.ARBITRUM_ONE,
+            event_type=HistoryEventType.STAKING,
+            event_subtype=HistoryEventSubType.REMOVE_ASSET,
+            asset=A_GRT_ARB,
+            balance=Balance(amount=FVal(withdrawn_amount)),
+            location_label=ADDY_USER_3_ARB,
+            notes=(
+                f'Withdraw {withdrawn_amount} GRT from indexer {indexer_address}'
             ),
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/decoders/test_thegraph.py:353]:
```suggestion
            notes=f'Withdraw {withdrawn_amount} GRT from indexer {indexer_address}',
```

File: rotkehlchen/chain/evm/decoding/thegraph/decoder.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    # -- DecoderInterface methods
+
+    @staticmethod
+    def counterparties() -> tuple[CounterpartyDetails, ...]:
+        return (THEGRAPH_CPT_DETAILS,)
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {
+            self.staking_contract: (self._decode_delegator_staking,),
+        }
```
Comment (yabirgb) [inline rotkehlchen/chain/evm/decoding/thegraph/decoder.py:185]: if it fits in pne line keep it in one line

```suggestion
        return {self.staking_contract: (self._decode_delegator_staking,)}
```

File: rotkehlchen/tests/unit/decoders/test_thegraph.py [hunk 5]
```diff
@@ -163,14 +170,189 @@ def test_thegraph_delegated_withdrawn(database, ethereum_inquirer):
...
             event_subtype=HistoryEventSubType.REMOVE_ASSET,
             asset=A_GRT,
-            balance=Balance(amount=FVal('1003.70342593701668535')),
+            balance=Balance(amount=FVal(withdrawn_amount)),
             location_label=ADDY_USER,
             notes=(
-                'Withdraw 1003.70342593701668535 GRT'
-                ' from indexer 0x6125eA331851367716beE301ECDe7F38A7E429e7'
+                f'Withdraw {withdrawn_amount} GRT from indexer {indexer_address}'
+            ),
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/decoders/test_thegraph.py:177]: keep this note in one line

**Review by yabirgb — COMMENTED** [review]
Failed tests due to the missing cassettes

File: rotkehlchen/chain/evm/decoding/thegraph/decoder.py (hunk 2 above)
Comment (moxw) [reply rotkehlchen/chain/evm/decoding/thegraph/decoder.py:36]: Yes, the undelegate on arb is different than that of eth mainnet.

File: rotkehlchen/chain/evm/decoding/thegraph/decoder.py [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+        if delegator is None or self.base.is_tracked(delegator) is False:
+            return DEFAULT_DECODING_OUTPUT
+        indexer = hex_or_bytes_to_address(context.tx_log.topics[1])
+        tokens_amount = hex_or_bytes_to_int(context.tx_log.data[:32])
+        lock_timeout_secs = hex_or_bytes_to_int(context.tx_log.data[64:128])
+        tokens_amount_norm = token_normalized_value(
+            token_amount=tokens_amount,
+            token=self.token,
+        )
+        log.info(tokens_amount_norm)
```
Comment (yabirgb) [inline rotkehlchen/chain/evm/decoding/thegraph/decoder.py:133]: you left this log that is not useful in any way. Remove it

File: rotkehlchen/chain/evm/decoding/thegraph/decoder.py [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+            balance=Balance(),
+            location_label=delegator,
+            notes=(
+                f'Undelegate {tokens_amount_norm} GRT from indexer {indexer}.'
+                f' Lock expires in {lock_timeout_secs} seconds'
+            ),
+            counterparty=CPT_THEGRAPH,
+            address=context.tx_log.address,
+        )
+        log.info(event)
```
Comment (yabirgb) [inline rotkehlchen/chain/evm/decoding/thegraph/decoder.py:150]: This shouldn't be here either

File: rotkehlchen/chain/evm/decoding/thegraph/decoder.py [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+                        events_list=context.decoded_events,
+                    )
+                    return DecodingOutput(event=burn_event)
+                break
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_stake_locked(self, context: DecoderContext) -> DecodingOutput:
+        delegator = hex_or_bytes_to_address(context.tx_log.topics[2])
+        log.info(delegator)
```
Comment (yabirgb) [inline rotkehlchen/chain/evm/decoding/thegraph/decoder.py:123]: this is a leftover too

File: rotkehlchen/chain/evm/decoding/thegraph/decoder.py [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+        elif context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED_LOCKED:
+            return self._decode_stake_locked(context)
+        elif context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED_WITHDRAWN:
+            return self._decode_stake_withdrawn(context)
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_stake_delegated(self, context: DecoderContext) -> DecodingOutput:
+        deposit_event, burn_event = None, None
+        delegator = hex_or_bytes_to_address(context.tx_log.topics[2])
+        log.info(self.base.is_tracked(delegator))
```
Comment (yabirgb) [inline rotkehlchen/chain/evm/decoding/thegraph/decoder.py:70]: this is a leftover. Please be careful

**Review by yabirgb — COMMENTED** [review]
image
This question about sqldiff was not answered

File: rotkehlchen/chain/evm/decoding/thegraph/decoder.py [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+# example delegate() call: https://etherscan.io/tx/0x6ed3377db652151fb8e4794dd994a921a2d029ad317bd3f2a2916af239490fec
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/thegraph/decoder.py:32]: All these examples are in the tests, right? You can probably remove the comments then. They just pollute the file

File: rotkehlchen/tests/unit/decoders/test_thegraph.py [hunk 11]
```diff
@@ -163,14 +170,185 @@ def test_thegraph_delegated_withdrawn(database, ethereum_inquirer):
...
+def test_thegraph_delegate_arbitrum_one(database, arbitrum_one_inquirer):
+    tx_hash = deserialize_evm_tx_hash('0x3c846f305330969fb0ddb87c5ae411b4e9692f451a7ff3237b6f71020030c7d1')  # noqa: E501
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=arbitrum_one_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    timestamp = TimestampMS(1713366299000)
+    gas_fees, approve_amount, delegate_amount, delegate_tax = '0.000003483187128', '1.009574848602990389', '24.9745', '0.1255'  # noqa: E501
+    indexer_address = '0xE13840A2E92e0Cb17A246609b432D0fA2e418774'
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/test_thegraph.py:194]: Here and in other places in these tests.

```suggestion
    indexer_address = string_to_evm_address('0xE13840A2E92e0Cb17A246609b432D0fA2e418774')
```

**Review by LefterisJP — APPROVED** [review]
LGTM. Some small nitpicks for next PR.

**Pull request merged**